tests/__tests__/bytes.ava.js

```diff
@@ -0,0 +1,225 @@
+    let stateMap = new Map()
+    // viewState doesn't work, because it tries to convert key to utf-8 string, which is not
+    let state = await jsvm.viewStateRaw()
+    for (let {key, value} of state) {
+        stateMap.set(key, value)
+    }
+    console.log(await jsvm.viewState())
+
+    t.deepEqual(
+        stateMap.get(encodeStateKey('123')),
+        Buffer.from('456').toString('base64')
+    );
+    // pass in utf-8 string instead of bytes, key and value become empty 
+    t.deepEqual(
+        stateMap.get(encodeStateKey([0xe6, 0xb0, 0xb4])),
+        undefined
+    );
+    t.deepEqual(
+        stateMap.get(encodeStateKey([])),
+        ''
+    );
+});
+
+test('Storage read bytes tests', async t => {
+    const { jsvm, ali, testContract } = t.context.accounts;
+
+    await ali.call(jsvm, 'call_js_contract', encodeCall(testContract.accountId, 'storage_write_bytes', ''), { attachedDeposit: '100000000000000000000000' });
+    let r = await jsvm.viewRaw('view_js_contract', encodeCall(testContract.accountId, 'storage_read_ascii_bytes', ''))
+    console.log(r)
+    t.deepEqual(
+        r.result,
+        [100, 101, 102]
+    );
+
+    r = await jsvm.viewRaw('view_js_contract', encodeCall(testContract.accountId, 'storage_read_arbitrary_bytes_key_utf8_sequence_bytes_value', ''))
+    t.deepEqual(
+        r.result,
+        [0xe6, 0xb0, 0xb4]
+    );
+
+    r = await jsvm.viewRaw('view_js_contract', encodeCall(testContract.accountId, 'storage_read_utf8_sequence_bytes_key_arbitrary_bytes_value', ''))
+    t.deepEqual(
+        r.result,
+        [0x00, 'a'.charCodeAt(0), 'b'.charCodeAt(0)]
+    );
+});
+
+test('panic tests', async t => {
+    const { jsvm, ali, testContract } = t.context.accounts;
+    let r = await ali.callRaw(jsvm, 'call_js_contract', encodeCall(testContract.accountId, 'panic_test', ''));
+    t.deepEqual(
+        r.result.receipts_outcome[0].outcome.status.Failure.ActionError.kind.FunctionCallError.ExecutionError,
+        'Smart contract panicked: explicit guest panic'
+    );
+
+    r = await ali.callRaw(jsvm, 'call_js_contract', encodeCall(testContract.accountId, 'panic_ascii_test', ''));
+    t.deepEqual(
+        r.result.receipts_outcome[0].outcome.status.Failure.ActionError.kind.FunctionCallError.ExecutionError,
+        'Smart contract panicked: abc'
+    );
+
+    r = await ali.callRaw(jsvm, 'call_js_contract', encodeCall(testContract.accountId, 'panic_js_number', ''));
+    t.deepEqual(
+        r.result.receipts_outcome[0].outcome.status.Failure.ActionError.kind.FunctionCallError.ExecutionError,
+        'Smart contract panicked: 356'
+    );
+
+    r = await ali.callRaw(jsvm, 'call_js_contract', encodeCall(testContract.accountId, 'panic_js_undefined', ''));
+    t.deepEqual(
+        r.result.receipts_outcome[0].outcome.status.Failure.ActionError.kind.FunctionCallError.ExecutionError,
+        'Smart contract panicked: explicit guest panic'
```

Comment:
There's no way to tell if a argument is not passed, or it's passed but undefined, same as nodejs:
```
Welcome to Node.js v18.2.0.
Type ".help" for more information.
> function a(x) { console.log(x) }
undefined
> a(undefined)
undefined
undefined
> a()
undefined
undefined
```